src/sql/workbench/contrib/executionPlan/browser/executionPlanComparisonPropertiesView.ts

```diff
@@ -320,36 +321,61 @@ export class ExecutionPlanComparisonPropertiesView extends ExecutionPlanProperti
 	 * @param props Map of properties that will be organized.
 	 * @returns A new map with different values appearing at the top and similar values appearing at the bottom.
 	 */
-	public sortPropertiesByDisplayValueEquivalency(props: Map<string, TablePropertiesMapEntry>, sortProperties: (props: Map<string, TablePropertiesMapEntry>) => Map<string, TablePropertiesMapEntry>): Map<string, TablePropertiesMapEntry> {
-		let unequalProperties: Map<string, TablePropertiesMapEntry> = new Map();
-		let equalProperties: Map<string, TablePropertiesMapEntry> = new Map();
+	public sortPropertiesByDisplayValueEquivalency(props: any[]): any[] {
+		const [unequalProperties, equalProperties] = this.splitEqualFromUnequalProperties(props);
 
-		[...props.entries()].forEach(prop => {
-			const [rowKey, rowEntry] = prop;
-			const primaryProp = rowEntry.primaryProp;
-			const secondaryProp = rowEntry.secondaryProp;
+		const organizedProperties: any[] = [...unequalProperties];
 
-			if (primaryProp?.displayValue.localeCompare(secondaryProp?.displayValue) === 0) {
-				equalProperties.set(rowKey, rowEntry);
-			}
-			else {
-				unequalProperties.set(rowKey, rowEntry);
-			}
-		});
+		if (equalProperties.length > 0) {
+			const equivalentPropertiesRow = {};
+			equivalentPropertiesRow['name'] = equivalentPropertiesRowHeader;
+			equivalentPropertiesRow['expanded'] = false;
+			equivalentPropertiesRow['treeGridChildren'] = equalProperties;
 
-		unequalProperties = sortProperties(unequalProperties);
-		equalProperties = sortProperties(equalProperties);
+			organizedProperties.push(equivalentPropertiesRow);
+		}
 
-		let map: Map<string, TablePropertiesMapEntry> = new Map();
-		unequalProperties.forEach((v, k) => {
-			map.set(k, v);
-		});
+		return organizedProperties;
+	}
 
-		equalProperties.forEach((v, k) => {
-			map.set(k, v);
-		});
+	private splitEqualFromUnequalProperties(props: any[]): any[] {
```

Comment:
The return type could be made better - I'd suggest an object with equal/unequal properties containing the sub-arrays. (or if you really want to keep this it should be `[any[], any[]]`

Reply:
How about Slick.SlickData?

Reply:
I'd highly recommend you make your own type that extends Slick.SlickData if you're going to be working with the row data directly. This will let you define what properties you expect so you can have actual type checking and make this stuff a lot easier to work with.

e.g. https://github.com/Microsoft/azuredatastudio/blob/3cb2f552a6f004dccd153577aecfd75f3d1bacae/src/sql/workbench/contrib/jobManagement/browser/jobsView.component.ts#L42